### PR TITLE
Fishy Cleanup

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
Balance "Remove some fish, fix up some others"

#### Purpose of change
The fish were kind of a mess. Some were specified up to species level, some were generic 'Salmon'. Their sizes also didn't always reflect the real-world average for a particular species. Overall it was a mess I wanted to clean up.

#### Describe the solution
**REMOVAL**
- [x] Trout was removed on account of being a filler, nondescript trout while we had all other trout species from the region
- [x] Steelhead Trout was removed on account of being literally the same animal as the Rainbow Trout we already have
- [x] Chinook Salmon, Coho Salmon, White Bass, Longear Sunfish and Redear Sunfish were removed on account of not being present in New England according to iNaturalist

**NAME CHANGES**
- [x] Salmon -> Atlantic Salmon
- [x] Whitefish -> Lake Whitefish
- [x] Perch -> Yellow Perch
- [x] Sunfish -> Banded Sunfish
- [x] Pumpkinseed Sunfish -> Pumpkinseed
- [x] Calico Bass -> Black Crappie
- [x] Bullhead -> Brown Bullhead
- [x] Pike -> Northern Pike
- [ ] Pickerel -> Chain Pickerel
- [x] Carp -> European Carp
- [x] Freshwater Eel -> American Eel

**SIZE CHANGES**
- [x] Trout, Salmon and Whitefish
- [x] Bass
- [x] Perches
- [x] Sunfish and Crappie
- [x] Catfish
- [x] Pikes
- [x] Cyprinids
- [x] The three at the end of the list
- [x] Crayfish

Also changed the Fallfish' description to not reference Gar. They're not related to Gar unless we're talking about different fish I'm not aware of. They're cyprinids. Carp.

ALSO ran the dictionary file through an autosorter so things are alphabetical now

#### Describe alternatives you've considered
Changing the name of the Kokanee Salmon to Sockeye Salmon but both work, not changing the Pumpkinseed's name since it's also applicable but its just uncomfortably long right now.

#### Additional context
**CALCULATIONS**
The volume was calculated by taking the average length of the fish present on fishbase, rounding it, dividing it by either 3/4/5/6/10 depending on the fish given (sunfish and crayfish were divided by 3; perch, some longer basses, some cyprinids and a few others were divided by 5; pike and catfish were divided by 6; eel was divided by 10; rest should be by 4 unless I forgot something). The actual volume calculation for a pike with a length of 24 would then be 4x4x24. The weights were calculated proportionally to maximum weight given on the specific fishbase page unless specified otherwise (and in the case of trout and salmon, as I calculated them before I got this calculation set properly). Another thing of note is salmon/trout sizes are often all over the place. Maximum sizes differ by a ton from average sizes. A good example of a problem causer here is the Brook Trout, which some sources cite as being on average 30cm long while others say it's 65cm long. What I'm saying here is if you have experience fishing those then feel free to yell at me, as I've got none and have been basing this solely off calculations and what the internet told me.

**PRESENCE SOURCES**
https://www.inaturalist.org/observations?photos&place_id=52339&quality_grade=research&taxon_id=47178&view=species - this is the link highlighting all ray-finned fish present in New England. If a species was not present on this list but was in the game it was obsoleted.
https://www.inaturalist.org/taxa/47516-Oncorhynchus-mykiss - the iNaturalist page for the Rainbow Trout, in the taxonomy section on the bottom, scrolling down to English names will show the Rainbow and Steelhead Trout are the same fish species, hence Steelhead Trout being obsoleted.

**SIZE SOURCES**
<details>
<summary>Brown Trout:</summary>

https://www.fishbase.se/summary/Salmo-trutta.html

https://en.wikipedia.org/wiki/Brown_trout

</details>
<details>
<summary>Brook Trout</summary>

https://www.fishbase.se/summary/Salvelinus-fontinalis.html

https://en.wikipedia.org/wiki/Brook_trout

</details>
<details>
<summary>Lake Trout</summary>

https://www.fishbase.se/summary/Salvelinus-namaycush.html

https://en.wikipedia.org/wiki/Lake_trout

</details>
<details>
<summary>Rainbow Trout</summary>

https://www.fishbase.se/summary/Oncorhynchus-mykiss.html

https://en.wikipedia.org/wiki/Rainbow_trout

</details>
<details>
<summary>Atlantic Salmon</summary>

https://www.fishbase.se/summary/Salmo-salar.html

https://en.wikipedia.org/wiki/Atlantic_salmon

</details>
<details>
<summary>Kokanee Salmon</summary>

https://www.fishbase.se/summary/Oncorhynchus-nerka.html

https://en.wikipedia.org/wiki/Sockeye_salmon

</details>
<details>
<summary>Lake Whitefish</summary>

https://www.fishbase.se/summary/Coregonus-clupeaformis.html

https://en.wikipedia.org/wiki/Lake_whitefish

</details>
<details>
<summary>Largemouth Bass</summary>

https://www.fishbase.se/summary/Micropterus-salmoides.html

https://animaldiversity.org/accounts/Micropterus_salmoides/

</details>
<details>
<summary>Smallmouth Bass</summary>

https://www.fishbase.se/summary/Micropterus-dolomieu.html

https://animaldiversity.org/accounts/Micropterus_dolomieu/

</details>
<details>
<summary>Striped Bass</summary>

https://www.fishbase.se/summary/Morone-saxatilis.html

https://animaldiversity.org/accounts/Morone_saxatilis/

</details>
<details>
<summary>Yellow Perch</summary>

https://www.fishbase.se/summary/Perca-flavescens.html

https://animaldiversity.org/accounts/Perca_flavescens/

</details>
<details>
<summary>Walleye</summary>

https://www.fishbase.se/summary/Sander-vitreus.html

https://animaldiversity.org/accounts/Sander_vitreus/

**IMPORTANT**: The sizes didn't add up in here. With the same average length as the Smallmouth Bass, which is a far bulkier fish, Walleye ended up with more weight despite being far more slender. I modified the size for the Walleye to have slightly more volume assuming that in reality the average adult length is higher than what Fishbase puts it as (wouldn't be the first time Fishbase did this)

</details>
<details>
<summary>Banded Sunfish</summary>

https://www.fishbase.se/summary/Enneacanthus-obesus.html

**IMPORTANT**: Found very little data about this particular species. The weight is calculated proportionally from the related Bluegill.

</details>
<details>
<summary>Pumpkinseed</summary>

https://www.fishbase.se/summary/Lepomis-gibbosus.html

https://animaldiversity.org/accounts/Lepomis_gibbosus/

</details>
<details>
<summary>Bluegill</summary>

https://www.fishbase.se/summary/Lepomis-macrochirus.html

https://animaldiversity.org/accounts/Lepomis_macrochirus/

</details>
<details>
<summary>Redbreast Sunfish</summary>

https://www.fishbase.se/summary/Lepomis-auritus.html

**IMPORTANT**: Found very little data about this particular species. The weight is calculated proportionally from the related Bluegill.

</details>
<details>
<summary>Green Sunfish</summary>

https://www.fishbase.se/summary/Lepomis-cyanellus.html

https://animaldiversity.org/accounts/Lepomis_cyanellus/

</details>
<details>
<summary>Rock Bass</summary>

https://www.fishbase.se/summary/Ambloplites-rupestris.html

https://animaldiversity.org/accounts/Ambloplites_rupestris/

</details>
<details>
<summary>Black Crappie</summary>

https://www.fishbase.se/summary/Pomoxis-nigromaculatus.html

https://animaldiversity.org/accounts/Pomoxis_nigromaculatus/

</details>
<details>
<summary>Warmouth</summary>

https://www.fishbase.se/summary/Lepomis-gulosus.html

**IMPORTANT**: Found very little data about this particular species. The weight is calculated proportionally from the related Bluegill.

</details>
<details>
<summary>Brown Bullhead</summary>

https://www.fishbase.se/summary/Ameiurus-nebulosus.html

https://animaldiversity.org/accounts/Ameiurus_nebulosus/

</details>
<details>
<summary>Channel Catfish</summary>

https://www.fishbase.se/summary/Ictalurus-punctatus.html

https://animaldiversity.org/accounts/Ictalurus_punctatus/

</details>
<details>
<summary>White Catfish</summary>

https://www.fishbase.se/summary/Ameiurus-catus.html

**IMPORTANT**: Found very little data about this particular species. The weight is calculated proportionally from the related Brown Bullhead.

</details>
<details>
<summary>Northern Pike</summary>

https://www.fishbase.se/summary/Esox-lucius.html

https://animaldiversity.org/accounts/Esox_lucius/

</details>
<details>
<summary>Chain Pickerel</summary>

https://www.fishbase.se/summary/Esox-niger.html

https://animaldiversity.org/accounts/Esox_niger/

</details>
<details>
<summary>Muskellunge</summary>

https://www.fishbase.se/summary/Esox-masquinongy.html

https://animaldiversity.org/accounts/Esox_masquinongy/

</details>
<details>
<summary>White Sucker</summary>

https://www.fishbase.se/summary/Catostomus-commersonii.html

https://animaldiversity.org/accounts/Catostomus_commersonii/

</details>
<details>
<summary>European Carp</summary>

https://www.fishbase.se/summary/Cyprinus-carpio.html

https://animaldiversity.org/accounts/Cyprinus_carpio/

**IMPORTANT**: This is the only fish from this list I have firsthand experience fishing and I can guarantee the carp you fish out would not be the size fishbase considers average. It's one of those fish where the average size and average adult size are not the same thing and so I made the European Carp larger than what is considered average by Fishbase.

</details>
<details>
<summary>Grass Carp</summary>

https://www.fishbase.se/summary/Ctenopharyngodon-idella.html

**IMPORTANT**: _Somehow_ Animal Diversity doesn't have a site for the Grass Carp so I used solely Fishbase for this one. Having said that, there's a red herring here. Yet again the average size and average adult size are not the same thing. Instead I used the maturity range Fishbase provided and assumed an 80cm long fish.

</details>
<details>
<summary>Bowfin</summary>

https://www.fishbase.se/summary/Amia-calva.html

https://animaldiversity.org/accounts/Amia_calva/

</details>
<details>
<summary>Fallfish</summary>

https://www.fishbase.se/summary/Semotilus-corporalis.html

https://animaldiversity.org/accounts/Semotilus_corporalis/

</details>
<details>
<summary>American Eel</summary>

https://www.fishbase.se/summary/Anguilla-rostrata.html

https://animaldiversity.org/accounts/Anguilla_rostrata/

</details>
<details>
<summary>Crayfish</summary>

https://www.infish.com.pl/wydawnictwo/Archives/Fasc/work_pdf/Vol26Fasc2/Vol26Fasc2_w06.pdf

**IMPORTANT**: This article highlights the size for the Signal Crayfish in Poland, which I used because I am kind of stupid. Initially the mutant crayfish referenced the Signal Crayfish in their description and I didn't think to check if the species is present in New England until after I was done with resizing. What I did is I used the calculations for this species and then calculated proportionally how much larger would the most common crayfish in New England, _Faxonius virilis_, be. Yes, it's an ass backwards way. It works though.

</details>

### CODE REVIEW

File: data/json/monsters/fish.json [hunk 1]
```diff
@@ -257,8 +242,9 @@
     "type": "MONSTER",
     "copy-from": "mon_fish_small",
     "name": { "str": "brook trout" },
-    "looks_like": "mon_fish_trout",
```
Comment (Fris0uman) [inline data/json/monsters/fish.json:260]: please replace the looks_like with something

File: data/json/monsters/fish.json [hunk 2]
```diff
@@ -221,28 +221,13 @@
...
-    "upgrades": { "half_life": 14, "into": "mon_fish_trout" },
-    "name": { "str_sp": "fry" }
-  },
   {
     "id": "mon_fish_brown_trout",
     "type": "MONSTER",
-    "copy-from": "mon_fish_medium",
+    "copy-from": "mon_fish_large",
     "name": { "str": "brown trout" },
-    "looks_like": "mon_fish_trout",
```
Comment (Fris0uman) [inline data/json/monsters/fish.json:244]: please replace the looks_like with something

File: data/json/monsters/fish.json [hunk 3]
```diff
@@ -273,8 +259,8 @@
     "type": "MONSTER",
     "copy-from": "mon_fish_large",
     "name": { "str": "lake trout" },
-    "looks_like": "mon_fish_trout",
```
Comment (Fris0uman) [inline data/json/monsters/fish.json:276]: same

File: data/json/monsters/fish.json [hunk 4]
```diff
@@ -287,10 +273,11 @@
   {
     "id": "mon_fish_rainbow_trout",
     "type": "MONSTER",
-    "copy-from": "mon_fish_medium",
+    "copy-from": "mon_fish_large",
     "name": { "str": "rainbow trout" },
-    "looks_like": "mon_fish_trout",
```
Comment (Fris0uman) [inline data/json/monsters/fish.json:292]: same

**Review by Fris0uman — COMMENTED** [review]
Since you used sources for those changes it would be nice to link directly to the pages you looked at. That way it'd be easier to double check

**Comment by Karol1223** [comment]
> Since you used sources for those changes it would be nice to link directly to the pages you looked at. That way it'd be easier to double check

Do you mean... you want me to post 3-r links per species for each species present? Because I can do that, sure, but I feel like that's kind of insanity. I'd propose linking the websites instead of the pages so that if someone wants to double-check me they can but have 4 links instead of overwhelming 60. Searching the species on the specific page should not be hard either.

Just let me know which approach to go with and I'll do so but I think a link to each individual page for each species is a bit much.

**Pull request merged**